rust/gui-client/src-tauri/src/client/gui.rs

```diff
@@ -253,6 +248,7 @@ pub(crate) fn run(
                 let app_handle = app.handle();
                 let _ctlr_task = tokio::spawn(async move {
                     let app_handle_2 = app_handle.clone();
+                    let telemetry_2 = telemetry.clone();
```

Comment:
A tricky you can do in these situations is to make a new scope and shadow the old variable:

```rust
tokio::spawn({
	let app_handle = app_handle.clone();
	let telemetry = telemetry.clone();

	async move {
		// ...
	}
})
```

Using a new scope avoids having to append `_2` or other suffixes to the variable.

Reply:
Another thing you could do here is construct `run_controller` first (without calling `await`). Then you can just call `.clone()` as part of calling the function.

Actually, you might not even need the `async move` block?

Would this work?

```rust
tokio::spawn(run_controller(...));
```